File: client-app/src/types/event.ts
export interface EventAttendee {
  id: string;
  displayName: string;
  imageUrl?: string | null;
}

export interface Event {
  id: string;
  title: string;
  startDate: string;
  endDate: string;
  description: string;
  category: string;
  isCancelled: boolean;
  city: string;
  venue: string;
  latitude: number;
  longitude: number;
  hostId: string;
  hostDisplayName: string;
  imageUrl: string | null;
  isHost: boolean;
  isGoing: boolean;
  attendees: EventAttendee[];
}

export interface AttendeeDto {
  id: string;
  displayName: string;
  imageUrl?: string;
  isHost: boolean;
}

export interface EventDetails extends Event {
  attendees: AttendeeDto[];
}

export interface PagedList<T> {
  items: T[];
  pageNumber: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;
}

export interface GetEventsResult {
  pagedEvents: PagedList<Event>;
  currentDate: string;
}

export interface EventParams {
  pageNumber?: number;
  pageSize?: number;
  startDate?: string;
  category?: number;
}

File: Domain/Event.cs
using System;

namespace Domain;

public class Event
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public required string Title { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public required string Description { get; set; }
    public required string Category { get; set; }
    public bool isCancelled { get; set; }

    public required string City { get; set; }
    public required string Venue { get; set; }
    public double Latitude { get; set; }
    public double Longitude { get; set; }
    
    public string? ImageUrl { get; set; }
    public string? PublicId { get; set; }

    //navigation properties

    public ICollection<EventAttendee> Attendees { get; set; } = [];
    public ICollection<EventComment> Comments { get; set; } = [];
}
